### PR TITLE
chore: unit tests to raise lightspeed utils coverage

### PR DESCRIPTION
**Pull request by omaciel**
## Summary

Continues the SonarQube coverage push for lightspeed code (follows #2882, #2884, #2893). Adds/extends vitest unit tests for seven `src/features/lightspeed/` utility files that were below 90% coverage, targeting the specific uncovered branches reported by SonarQube.

| File | Before |
|---|---|
| `lightspeed/explorerWebviewViewProvider.ts` | 44% |
| `lightspeed/utils/watchers.ts` | 50% |
| `lightspeed/utils/errors.ts` | 85% |
| `lightspeed/utils/webUtils.ts` | 88% |
| `lightspeed/utils/readVarFiles.ts` | 88% |
| `lightspeed/utils/explanationUtils.ts` | 89% |
| `features/utils/lightspeed.ts` | 78% |

## What's covered
- **explorerWebviewViewProvider**: `onDidDispose` cleanup loop (incl. falsy-entry skip) and `refreshWebView` resolved/unresolved.
- **watchers**: `onDidChange`/`onDidDelete`/`onDidCreate` callbacks across workspace-folder, standard-roles, and cache-hit/miss permutations.
- **errors**: `formatErrorDetail` for every value type (object/number/boolean/bigint/symbol/named+anonymous function), `HTTPError`, `isError`.
- **webUtils**: `generateCodeVerifier` recursion, `coerceExpiresIn` table, `getUserTypeLabel`, `getLoggedInUserDetails`.
- **readVarFiles**: missing-file early return + Error vs non-Error catch branches.
- **explanationUtils**: `getObjectKeys` / `isDocumentInRole` / `isPlaybook`.
- **features/utils/lightspeed**: `adjustInlineSuggestionIndent` indent + malformed-line branches.

## Testing
Test-only change — no production source modified. New/extended files: **92 passing**.

```
npx vitest run --project ext test/unit/lightspeed/... test/unit/utils/lightspeed.test.ts
# Test Files 7 passed (7) · Tests 92 passed (92)
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Added Vitest unit tests to raise Lightspeed utilities coverage by mocking VS Code APIs, filesystem/YAML, crypto, and helpers. The tests exercise previously uncovered branches for webview lifecycle, role watcher cache/workspace guards, error formatting/HTTPError behavior, code-verifier and expiry coercion edge cases, var-file read failure paths, YAML role/document/playbook checks, and inline suggestion indentation/format cleanup.

- Expanded webview provider tests for disposal cleanup and `refreshWebView()` messaging/no-op behavior
- Strengthened watcher tests across create/change/delete flows, including missing-cache and absent `workspaceFolders` scenarios
- Added utilities tests for error formatting, `readVarFiles` missing/throw/parse behaviors, explanation/web helpers, and inline suggestion indentation handling

chore: unit tests to raise lightspeed utils coverage
Hrithik Gavankar
GitHub

### CODE REVIEW

File: test/unit/lightspeed/utils/watchers.test.ts [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        isFile: () => false,
+      } as never);
+      (vscode.workspace as { workspaceFolders: unknown }).workspaceFolders = [
+        { uri: { fsPath: "/ws" } },
+      ];
+
+      expect(() =>
+        handlers.delete?.({ fsPath: "/ws/roles/missing" }),
+      ).not.toThrow();
+    });
```
[the remark this answers is not in the document]
Comment (omaciel) [reply test/unit/lightspeed/utils/watchers.test.ts:233]: Good catch — this was sharper than a test gap. `dirPath in ansibleRolesCache[workspaceFolder]` actually threw a `TypeError` (`in` on `undefined`) when the workspace had no cache bucket. Fixed the handler with a guard and added a regression test (`does not throw when the workspace has no cache bucket`). Done in d671078.

**Comment by omaciel** [comment]
@coderabbitai all three review items addressed in d671078:

1. **watchers.test.ts (Major)** — confirmed the missing-bucket path actually threw `TypeError: Cannot use 'in' operator … in undefined`. Guarded the `onDidDelete` handler (`ansibleRolesCache[workspaceFolder] && dirPath in …`) and added a `does not throw when the workspace has no cache bucket` regression test.
2. **readVarFiles.test.ts (nitpick)** — now asserts the literal `"key: value\n"` and verifies `yaml.parse` was called with `{ keepSourceTokens: true }` (plus `yaml.stringify` called), instead of recomputing the pipeline.
3. **utils/lightspeed.test.ts (nitpick)** — spy on `console.error` in the malformed-line test to keep output clean and assert the branch explicitly.

Note: the failing CI checks here are an unrelated repo-wide `task uv` (Python sync) setup failure also affecting the sibling PRs — not from this diff.

**Review by Hrithik-Gavankar — APPROVED** [review]
